File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int i = 0, len = nums.size(), maxx = 0;
        if(len == 1) return true;
        for(; i<len; i++){
            if(i > maxx) return false;
            maxx = max(maxx, i+nums[i]);
            if(maxx >= len - 1) return true;
        }
        return false;
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int i = 0, len = nums.length, max = 0;
        if(len == 1) return true;
        for(; i<len; i++){
            if(i > max) return false;
            max = Math.max(max, i+nums[i]);
            if(max >= len - 1) return true;
        }
        return false;
    }
}